<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>PerceptronWindow</class>
 <widget class="QMainWindow" name="PerceptronWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>904</width>
    <height>713</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>904</width>
    <height>643</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>AI Fundamentals - 221481 - Piotr Luczak</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QVBoxLayout" name="verticalLayout_7">
    <item>
     <widget class="QTabWidget" name="menuTab">
      <property name="minimumSize">
       <size>
        <width>753</width>
        <height>323</height>
       </size>
      </property>
      <property name="toolTip">
       <string/>
      </property>
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="singleNeuronTab">
       <attribute name="title">
        <string>Single Neuron</string>
       </attribute>
       <layout class="QHBoxLayout" name="horizontalLayout_2">
        <item>
         <widget class="QGroupBox" name="inputGroup">
          <property name="minimumSize">
           <size>
            <width>250</width>
            <height>278</height>
           </size>
          </property>
          <property name="title">
           <string>Input values</string>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout">
           <item>
            <widget class="QTableWidget" name="inputTable">
             <property name="verticalScrollBarPolicy">
              <enum>Qt::ScrollBarAlwaysOn</enum>
             </property>
             <property name="horizontalScrollBarPolicy">
              <enum>Qt::ScrollBarAsNeeded</enum>
             </property>
             <property name="sizeAdjustPolicy">
              <enum>QAbstractScrollArea::AdjustIgnored</enum>
             </property>
             <property name="alternatingRowColors">
              <bool>true</bool>
             </property>
             <property name="selectionMode">
              <enum>QAbstractItemView::NoSelection</enum>
             </property>
             <property name="selectionBehavior">
              <enum>QAbstractItemView::SelectColumns</enum>
             </property>
             <property name="rowCount">
              <number>0</number>
             </property>
             <property name="columnCount">
              <number>2</number>
             </property>
             <attribute name="horizontalHeaderCascadingSectionResizes">
              <bool>true</bool>
             </attribute>
             <attribute name="horizontalHeaderDefaultSectionSize">
              <number>110</number>
             </attribute>
             <attribute name="horizontalHeaderStretchLastSection">
              <bool>true</bool>
             </attribute>
             <attribute name="verticalHeaderCascadingSectionResizes">
              <bool>false</bool>
             </attribute>
             <column>
              <property name="text">
               <string>Inputs</string>
              </property>
             </column>
             <column>
              <property name="text">
               <string>Weights</string>
              </property>
             </column>
            </widget>
           </item>
           <item>
            <layout class="QHBoxLayout" name="inputButtonsLayout">
             <item>
              <widget class="QPushButton" name="addRowButton">
               <property name="text">
                <string>Add row</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="removeRowButton">
               <property name="text">
                <string>Remove row</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <layout class="QVBoxLayout" name="centralLayout">
          <item alignment="Qt::AlignHCenter|Qt::AlignTop">
           <widget class="QGroupBox" name="simulationBox">
            <property name="minimumSize">
             <size>
              <width>183</width>
              <height>0</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>16777215</width>
              <height>160</height>
             </size>
            </property>
            <property name="title">
             <string>Simulation</string>
            </property>
            <layout class="QFormLayout" name="formLayout_4">
             <item row="0" column="0">
              <widget class="QComboBox" name="functionBox">
               <property name="minimumSize">
                <size>
                 <width>75</width>
                 <height>23</height>
                </size>
               </property>
              </widget>
             </item>
             <item row="0" column="1">
              <widget class="QLabel" name="functionLabel">
               <property name="text">
                <string>Sigmoid function</string>
               </property>
              </widget>
             </item>
             <item row="1" column="0">
              <widget class="QDoubleSpinBox" name="betaBox">
               <property name="minimumSize">
                <size>
                 <width>75</width>
                 <height>23</height>
                </size>
               </property>
               <property name="buttonSymbols">
                <enum>QAbstractSpinBox::NoButtons</enum>
               </property>
               <property name="decimals">
                <number>4</number>
               </property>
              </widget>
             </item>
             <item row="1" column="1">
              <widget class="QLabel" name="logisticLabel">
               <property name="text">
                <string>Logistic beta</string>
               </property>
              </widget>
             </item>
             <item row="2" column="0">
              <widget class="QDoubleSpinBox" name="thetaBox">
               <property name="minimumSize">
                <size>
                 <width>75</width>
                 <height>23</height>
                </size>
               </property>
               <property name="buttonSymbols">
                <enum>QAbstractSpinBox::NoButtons</enum>
               </property>
               <property name="decimals">
                <number>4</number>
               </property>
               <property name="minimum">
                <double>-10.000000000000000</double>
               </property>
               <property name="maximum">
                <double>10.000000000000000</double>
               </property>
               <property name="singleStep">
                <double>0.100000000000000</double>
               </property>
              </widget>
             </item>
             <item row="2" column="1">
              <widget class="QLabel" name="thetaLabel">
               <property name="text">
                <string>Theta</string>
               </property>
              </widget>
             </item>
             <item row="3" column="0" colspan="2">
              <widget class="QPushButton" name="calculateButton">
               <property name="text">
                <string>Calculate</string>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
          <item alignment="Qt::AlignHCenter|Qt::AlignBottom">
           <widget class="QGroupBox" name="trainingBox">
            <property name="minimumSize">
             <size>
              <width>173</width>
              <height>0</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>16777215</width>
              <height>190</height>
             </size>
            </property>
            <property name="title">
             <string>Training</string>
            </property>
            <layout class="QFormLayout" name="formLayout_5">
             <item row="0" column="0">
              <widget class="QDoubleSpinBox" name="expectedBox">
               <property name="minimumSize">
                <size>
                 <width>75</width>
                 <height>23</height>
                </size>
               </property>
               <property name="buttonSymbols">
                <enum>QAbstractSpinBox::NoButtons</enum>
               </property>
               <property name="decimals">
                <number>4</number>
               </property>
               <property name="minimum">
                <double>-10.000000000000000</double>
               </property>
               <property name="maximum">
                <double>10.000000000000000</double>
               </property>
               <property name="singleStep">
                <double>0.100000000000000</double>
               </property>
              </widget>
             </item>
             <item row="0" column="1">
              <widget class="QLabel" name="expectedLabel">
               <property name="text">
                <string>Expected</string>
               </property>
              </widget>
             </item>
             <item row="1" column="0">
              <widget class="QDoubleSpinBox" name="epsilonBox">
               <property name="minimumSize">
                <size>
                 <width>75</width>
                 <height>23</height>
                </size>
               </property>
               <property name="buttonSymbols">
                <enum>QAbstractSpinBox::NoButtons</enum>
               </property>
               <property name="decimals">
                <number>4</number>
               </property>
               <property name="minimum">
                <double>-10.000000000000000</double>
               </property>
               <property name="maximum">
                <double>10.000000000000000</double>
               </property>
               <property name="singleStep">
                <double>0.100000000000000</double>
               </property>
              </widget>
             </item>
             <item row="1" column="1">
              <widget class="QLabel" name="epsilonLabel">
               <property name="text">
                <string>Epsilon</string>
               </property>
              </widget>
             </item>
             <item row="2" column="0">
              <widget class="QDoubleSpinBox" name="learningRateBox">
               <property name="minimumSize">
                <size>
                 <width>75</width>
                 <height>23</height>
                </size>
               </property>
               <property name="buttonSymbols">
                <enum>QAbstractSpinBox::NoButtons</enum>
               </property>
               <property name="decimals">
                <number>4</number>
               </property>
               <property name="minimum">
                <double>-10.000000000000000</double>
               </property>
               <property name="maximum">
                <double>10.000000000000000</double>
               </property>
               <property name="singleStep">
                <double>0.100000000000000</double>
               </property>
              </widget>
             </item>
             <item row="2" column="1">
              <widget class="QLabel" name="etaLabel">
               <property name="text">
                <string>Eta</string>
               </property>
              </widget>
             </item>
             <item row="3" column="0">
              <widget class="QSpinBox" name="maxIterBox">
               <property name="minimumSize">
                <size>
                 <width>75</width>
                 <height>23</height>
                </size>
               </property>
               <property name="buttonSymbols">
                <enum>QAbstractSpinBox::UpDownArrows</enum>
               </property>
               <property name="value">
                <number>99</number>
               </property>
              </widget>
             </item>
             <item row="3" column="1">
              <widget class="QLabel" name="iterationsLabel">
               <property name="text">
                <string>Max iterations</string>
               </property>
              </widget>
             </item>
             <item row="4" column="0" colspan="2">
              <widget class="QPushButton" name="trainButton">
               <property name="text">
                <string>Train</string>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QChartView" name="sigmoidView">
          <property name="minimumSize">
           <size>
            <width>360</width>
            <height>360</height>
           </size>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="networkTab">
       <attribute name="title">
        <string>Neural Network</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_3">
        <item row="1" column="2">
         <widget class="QGroupBox" name="networkTrainingGroup">
          <property name="title">
           <string>Training</string>
          </property>
          <layout class="QFormLayout" name="formLayout_2">
           <item row="0" column="0">
            <widget class="QDoubleSpinBox" name="networkEtaBox">
             <property name="minimumSize">
              <size>
               <width>75</width>
               <height>23</height>
              </size>
             </property>
             <property name="buttonSymbols">
              <enum>QAbstractSpinBox::NoButtons</enum>
             </property>
             <property name="decimals">
              <number>4</number>
             </property>
             <property name="maximum">
              <double>2.000000000000000</double>
             </property>
             <property name="singleStep">
              <double>0.010000000000000</double>
             </property>
             <property name="value">
              <double>0.010000000000000</double>
             </property>
            </widget>
           </item>
           <item row="0" column="1">
            <widget class="QLabel" name="etaLabel_2">
             <property name="text">
              <string>Learning rate</string>
             </property>
            </widget>
           </item>
           <item row="2" column="0">
            <widget class="QSpinBox" name="networkIterationsBox">
             <property name="minimumSize">
              <size>
               <width>75</width>
               <height>23</height>
              </size>
             </property>
             <property name="buttonSymbols">
              <enum>QAbstractSpinBox::NoButtons</enum>
             </property>
             <property name="maximum">
              <number>1000</number>
             </property>
             <property name="value">
              <number>99</number>
             </property>
            </widget>
           </item>
           <item row="2" column="1">
            <widget class="QLabel" name="iterationsLabel_2">
             <property name="text">
              <string>Epochs</string>
             </property>
            </widget>
           </item>
           <item row="1" column="0">
            <widget class="QDoubleSpinBox" name="networkEpsilonBox">
             <property name="minimumSize">
              <size>
               <width>75</width>
               <height>23</height>
              </size>
             </property>
             <property name="buttonSymbols">
              <enum>QAbstractSpinBox::NoButtons</enum>
             </property>
             <property name="decimals">
              <number>4</number>
             </property>
             <property name="maximum">
              <double>2.000000000000000</double>
             </property>
             <property name="singleStep">
              <double>0.100000000000000</double>
             </property>
             <property name="value">
              <double>0.001000000000000</double>
             </property>
            </widget>
           </item>
           <item row="1" column="1">
            <widget class="QLabel" name="epsilonLabel_2">
             <property name="text">
              <string>Epsilon</string>
             </property>
            </widget>
           </item>
           <item row="4" column="0" colspan="2">
            <widget class="QPushButton" name="networkTrainButton">
             <property name="text">
              <string>Train</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item row="2" column="2">
         <widget class="QGroupBox" name="networkTestingGroup">
          <property name="title">
           <string>Testing</string>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_5">
           <item>
            <widget class="QPushButton" name="networkCalculateButton">
             <property name="text">
              <string>Calculate</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item row="0" column="2">
         <widget class="QGroupBox" name="networkLogisticGroup">
          <property name="title">
           <string>Logistic function parameters</string>
          </property>
          <layout class="QFormLayout" name="formLayout_3">
           <item row="1" column="0">
            <widget class="QDoubleSpinBox" name="networkBetaBox">
             <property name="minimumSize">
              <size>
               <width>75</width>
               <height>23</height>
              </size>
             </property>
             <property name="buttonSymbols">
              <enum>QAbstractSpinBox::NoButtons</enum>
             </property>
             <property name="decimals">
              <number>4</number>
             </property>
             <property name="value">
              <double>1.000000000000000</double>
             </property>
            </widget>
           </item>
           <item row="2" column="0">
            <widget class="QDoubleSpinBox" name="networkThetaBox">
             <property name="enabled">
              <bool>false</bool>
             </property>
             <property name="minimumSize">
              <size>
               <width>75</width>
               <height>23</height>
              </size>
             </property>
             <property name="buttonSymbols">
              <enum>QAbstractSpinBox::NoButtons</enum>
             </property>
             <property name="decimals">
              <number>4</number>
             </property>
             <property name="value">
              <double>1.000000000000000</double>
             </property>
            </widget>
           </item>
           <item row="1" column="1">
            <widget class="QLabel" name="betaLabel">
             <property name="text">
              <string>Beta</string>
             </property>
            </widget>
           </item>
           <item row="3" column="0" colspan="2">
            <widget class="QPushButton" name="networkCreateButton">
             <property name="text">
              <string>Create Network</string>
             </property>
            </widget>
           </item>
           <item row="2" column="1">
            <widget class="QLabel" name="thetaLabel_2">
             <property name="text">
              <string>Theta (unused)</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QGroupBox" name="networkInputGroup">
          <property name="title">
           <string>Network configuration</string>
          </property>
          <layout class="QFormLayout" name="formLayout_6">
           <item row="1" column="0">
            <widget class="QSpinBox" name="networkInputsBox">
             <property name="minimum">
              <number>1</number>
             </property>
             <property name="maximum">
              <number>999999999</number>
             </property>
             <property name="value">
              <number>5</number>
             </property>
            </widget>
           </item>
           <item row="1" column="1">
            <widget class="QLabel" name="networkInputsLabel">
             <property name="text">
              <string>Dimensions (input nodes)</string>
             </property>
            </widget>
           </item>
           <item row="2" column="0">
            <widget class="QSpinBox" name="networkClassesBox">
             <property name="minimum">
              <number>2</number>
             </property>
             <property name="maximum">
              <number>999999999</number>
             </property>
             <property name="value">
              <number>4</number>
             </property>
            </widget>
           </item>
           <item row="3" column="0">
            <widget class="QSpinBox" name="networkSamplesBox">
             <property name="minimum">
              <number>1</number>
             </property>
             <property name="maximum">
              <number>999999999</number>
             </property>
             <property name="value">
              <number>200</number>
             </property>
            </widget>
           </item>
           <item row="2" column="1">
            <widget class="QLabel" name="networkOutputsLabel">
             <property name="text">
              <string>Classes (Output nodes)</string>
             </property>
            </widget>
           </item>
           <item row="3" column="1">
            <widget class="QLabel" name="networkSamplesLabel">
             <property name="text">
              <string>Samples</string>
             </property>
            </widget>
           </item>
           <item row="4" column="1">
            <widget class="QLabel" name="networkSplitLabel">
             <property name="text">
              <string>Train - Test split</string>
             </property>
            </widget>
           </item>
           <item row="4" column="0">
            <widget class="QDoubleSpinBox" name="networkSplitRatioBox">
             <property name="minimum">
              <double>0.010000000000000</double>
             </property>
             <property name="maximum">
              <double>1.000000000000000</double>
             </property>
             <property name="singleStep">
              <double>0.100000000000000</double>
             </property>
             <property name="stepType">
              <enum>QAbstractSpinBox::DefaultStepType</enum>
             </property>
             <property name="value">
              <double>0.700000000000000</double>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QGroupBox" name="networkIntermediateGroup">
          <property name="title">
           <string>Intermediate layers</string>
          </property>
          <layout class="QFormLayout" name="formLayout">
           <item row="0" column="0">
            <widget class="QSpinBox" name="networkLayerBox">
             <property name="minimumSize">
              <size>
               <width>75</width>
               <height>23</height>
              </size>
             </property>
            </widget>
           </item>
           <item row="0" column="1">
            <widget class="QLabel" name="layerCountLabel">
             <property name="text">
              <string>Number of layers</string>
             </property>
            </widget>
           </item>
           <item row="1" column="0">
            <widget class="QSpinBox" name="networkNeuronBox">
             <property name="minimumSize">
              <size>
               <width>75</width>
               <height>23</height>
              </size>
             </property>
             <property name="value">
              <number>3</number>
             </property>
            </widget>
           </item>
           <item row="1" column="1">
            <widget class="QLabel" name="neuronCountLabel">
             <property name="text">
              <string>Neurons per layer</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
    <item>
     <widget class="QTextBrowser" name="outputText">
      <property name="minimumSize">
       <size>
        <width>753</width>
        <height>192</height>
       </size>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>QChartView</class>
   <extends>QGraphicsView</extends>
   <header>QtCharts</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
